web/src/components/dashboards/usePerses.ts

```diff
@@ -0,0 +1,99 @@
+import { PersesDashboardMetadata } from './perses-client';
+import { useCallback, useReducer, useRef } from 'react';
+import { fetchPersesDashboardsMetadata } from './perses-client';
+
+const isAbortError = (error: unknown): boolean =>
+  error instanceof Error && error.name === 'AbortError';
+
+type State = {
+  dashboardsData: PersesDashboardMetadata[];
+  isLoadingDashboardsData: boolean;
+  dashboardsError: unknown;
+};
+
+type Action =
+  | {
+      type: 'dashboardsMetadataResponse';
+      payload: { dashboardsData: PersesDashboardMetadata[] };
+    }
+  | {
+      type: 'dashboardsRequest';
+    }
+  | {
+      type: 'dashboardsError';
+      payload: { error: unknown };
+    };
+
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case 'dashboardsMetadataResponse':
+      return {
+        ...state,
+        isLoadingDashboardsData: false,
+        dashboardsData: action.payload.dashboardsData,
+      };
+    case 'dashboardsError':
+      return {
+        ...state,
+        isLoadingDashboardsData: false,
+        dashboardsError: action.payload.error,
+      };
+    case 'dashboardsRequest':
+      return {
+        ...state,
+        isLoadingDashboardsData: true,
+        dashboardsData: undefined,
+        dashboardsError: undefined,
+      };
+    default:
+      return state;
+  }
+};
+
+export const usePerses = () => {
```

Comment:
We should change this hook name to be something like "usePersesDashboardList", as there are other perses hook we will need to create in the future

Reply:
I'm following the structure set by https://github.com/openshift/logging-view-plugin/blob/1319d70979d33cfa936dae6890c7bb02d1fa961d/web/src/hooks/useLogs.ts#L4 per our meeting on Thursday. 

I believe the other Perses hooks we create in the feature will live within usePerses.